File: Clase0606/persistencia/Medicamento.java
package Clase0606.persistencia;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class Medicamento {
    private String rutaBase = "F:/program/aux_121/Programacion-2-I2025/Clase0606/medicamentosJava/";

    private int id;
    private String nombre;
    private String tipo;
    private double precio;
    private int stock;

    private String archivo;

    public Medicamento(int id, String nombre, String tipo, double precio, int stock) {
        this.id = id;
        this.nombre = nombre;
        this.tipo = tipo;
        this.precio = precio;
        this.stock = stock;

        this.archivo = rutaBase + "medicamento"+ this.id + ".txt";
    }

    public Medicamento(String rutaArchivo) throws IOException{
        this.archivo = rutaArchivo;
        File archivo = new File(rutaArchivo);

        if(! archivo.exists()){
            throw new FileNotFoundException("El archivo no existe: persona" +this.id+".txt" );
        }else{
            BufferedReader br = new BufferedReader(new FileReader(archivo));

            this.id = Integer.parseInt(br.readLine());
            this.nombre = br.readLine();
            this.tipo = br.readLine();
            this.precio = Double.parseDouble(br.readLine());
            this.stock = Integer.parseInt(br.readLine());

            br.close();
        }
    }

    public void guardar() throws IOException{
        File carpeta = new File(rutaBase);

        if (!carpeta.exists()) {
            carpeta.mkdirs(); // Crea la carpeta si no existe
        }

        File archivo = new File(this.archivo);
        if(!archivo.exists()) {
            archivo.createNewFile(); // Crea el archivo si no existe
        }

        BufferedWriter bw = new BufferedWriter(new FileWriter(this.archivo));

        bw.write(this.id + "\n");
        bw.write(this.nombre + "\n");
        bw.write(this.tipo + "\n");
        bw.write(this.precio + "\n");
        bw.write(this.stock + "\n");

        bw.close();

    }

    public void actualizar() throws IOException{
        eliminar();
        guardar();
    }

    public void eliminar(){
        File f = new File(this.archivo);
        if (f.exists()) f.delete();
    }

    public String toString() {
        return "Medicamento{" +
                "id=" + id +
                ", nombre='" + nombre + '\'' +
                ", tipo='" + tipo + '\'' +
                ", precio=" + precio +
                ", stock=" + stock +
                '}';
    }

    public void setStock(int stock){
        this.stock = stock;
    }
    public String getNombre(){
        return this.nombre;
    }
    public int getStock(){
        return this.stock;
    }
    public double getPrecio(){
        return this.precio;
    }
}

File: Clase0606/Medicamento.py
import os
import json

rutaBase = "F:/program/aux_121/Programacion-2-I2025/Clase0606/medicamentoPython"

class Medicamento:
    def __init__(self, id=None, nombre=None, tipo = None, precio=None, stock=None):
        self.id = id
        self.nombre = nombre
        self.tipo = tipo
        self.precio = precio
        self.stock = stock
        self.archivo = f"{rutaBase}/medicamento{self.id}.json"
        
    def guardar(self):
        os.makedirs(rutaBase, exist_ok=True)
        
        if not os.path.exists(self.archivo):
            with open(self.archivo, 'w') as archivo:
                datos = {
                    "id": self.id,
                    "nombre": self.nombre,
                    "tipo": self.tipo,
                    "precio": self.precio,
                    "stock": self.stock
                }
                
                json.dump(datos, archivo, indent=4)
    
    def cargarDatos(self, rutaArchivo):
        self.archivo = rutaArchivo
        if os.path.isfile(self.archivo):
            with open(self.archivo, 'r') as archivo:
                datos = json.load(archivo)
                self.precio = datos.get("precio")
                self.id = datos.get("id")
                self.tipo = datos.get("tipo")
                self.nombre = datos.get("nombre")
                self.stock = datos.get("stock")
                print(f"Datos cargados: medicamento{self.id}.json")
        else:
            print(f"El archivo medicamento{self.id}.json no existe.")
            
    def actualizar(self):
        with open(self.archivo, 'r') as archivo:
            datos = json.load(archivo)
            datos["nombre"] = self.nombre
            datos["tipo"] = self.tipo
            datos["precio"] = self.precio
            datos["stock"] = self.stock
            with open(self.archivo, 'w') as archivo:
                json.dump(datos, archivo, indent=4)
    
    def eliminar(self):
        if os.path.isfile(self.archivo):
            os.remove(self.archivo)
            print(f"Archivo medicamento{self.id}.json eliminado.")
        else:
            print(f"El archivo medicamento{self.id}.json no existe.")
            
    def __str__(self):
        return f"ID: {self.id}, Nombre: {self.nombre}, Tipo: {self.tipo}, Precio: {self.precio}, Stock: {self.stock}"